cellmlmanip/units.py

```diff
@@ -362,30 +372,43 @@ def traverse(self, expr: sympy.Expr):
         """descends the Sympy expression and performs Pint unit arithmetic on sub-expressions
         :param expr: a Sympy expression
         :returns: the quantity (i.e. expression * unit) of the expression
-        :throws:
+        :throws: KeyError - if variable not found in metadata
+                 InvalidUnitsError - if expression just cannot calculate
+                 UnexpectedMathUnitsError - if math is not supported
+                 BooleanUnitsError - if math returns booleans
+                 InputArgumentsMustBeDimensionlessError - if input arguments should be dimensionless
+                 InputArgumentsInvalidUnitsError - if input arguments should have same units
+                 InputArgumentMustBeNumberError - if one of input arguments should be a number
+        NOTE: Sympy will throw exceptions if the expression is badly formed before we even get here
         """
         # collect the units for each argument of the expression (might be sub-expression)
         quantity_per_arg = []
 
-        # If the expression is a piecewise
+        # need to work out units of each child atom of the expression
+        # piecewise and derivatives are special cases
         if expr.is_Piecewise:
-            # Treat each sub-function an argument (& ignore the condition)
+            # Treat each sub-function an argument (& ignore the conditional statements)
             for arg in expr.args:
                 quantity_per_arg.append(self.traverse(arg[0]))
         elif expr.is_Derivative:
+            # In a derivative expression the first argument is function to be differentiated
+            # the second argument is a tuple of which the first is symbol of the differentiator(?)
```

Comment:
How bizarre!

Reply:
Yup! 🤷‍♂ Introduced in latest version of Sympy; second argument of second tuple is the _n_ for nth derivatives.